waku/waku_rln_relay/group_manager/on_chain/group_manager.nim

```diff
@@ -142,10 +144,8 @@ when defined(rln_v2):
     initializedGuard(g)
 
     # convert the rateCommitment struct to a leaf value
-    let leavesRes = rateCommitments.toLeaves()
-    if leavesRes.isErr():
-      raise newException(CatchableError, "failed to convert rateCommitments to leaves: " & leavesRes.error)
-    let leaves = cast[seq[seq[byte]]](leavesRes.get())
+    let leaves = rateCommitments.toLeaves().valueOr:
+      raise newException(ValueError, "failed to convert rateCommitments to leaves: " & $error)
```

Comment:
Unrelated to this PR but I'm more keen on avoid raising exceptions whenever is possible

Reply:
Will address in follow up prs!